### ISSUE TITLE
No name is provided for "checkboxes", "clear" and "New Task" button: A11y_.NET 9 template_.NET 9 template_Project_Name Role Value.

### ISSUE THREAD

**Issue by AnKushSingh05** [dotnet/maui#26816]
### GitHub Tags
#A11yTCS; #A11ySev2; #Win32; #WCAG4.1.2; #SH_.NET9template_MAC_Dec2024; #DesktopApp; #.NET Core; #; #A11yMAS; #Name Role Value; #MAC;

### Environment Details:
Application Name: .NET 9 template tests that include some 3rd party controls
MAC Version 15.2

### Repro Steps:

1. Install and open ".NET 9 template" app.
2. "Home Screen" will open.
3. Tab till checkboxes section.
4. Observe that No name is provided for "checkboxes", "clear" and "New Task" button.

**Similar issue is also observed for Change theme button**

1. Install and open ".NET 9 template" app.
2. "Home Screen" will open.
3. Tab till Hamburger menu and press enter.
4. Tab till theme button.
5. Observe that no name is given for dark/light theme.

**Similar issue is also observed for delete button**

1. Install and open ".NET 9 template" app.
2. . "Home Screen" will open.
3.  Tab till Hamburger menu and press enter.
4. Tab till Categories and tags and press enter.
5. Observe that no name is given for delete button.

### Actual Result:
No name is provided for "checkboxes", "clear" and "New Task" button.

### Expected Result:
Name should be provided for "checkboxes", "clear" and "New Task" button.

### User Impact:
if screen reader does not announce the name, then visually impaired user cannot find the purpose, use and action of the control.

### Attachment:

https://github.com/user-attachments/assets/2ef9e76b-4e0b-4208-bb60-bdb3bddcc792

<img width="1131" alt="Image" src="https://github.com/user-attachments/assets/64f1c48a-059d-4b02-86d8-e5f72f3ac821" />
<img width="1131" alt="Image" src="https://github.com/user-attachments/assets/4192fdea-7b69-4f74-ab8c-98ee54090a2d" />
<img width="1131" alt="Image" src="https://github.com/user-attachments/assets/bfe3b085-a705-480a-8c92-d9e5852dd20a" />
<img width="1098" alt="Image" src="https://github.com/user-attachments/assets/1cb3aaa6-11bb-4fd1-8e7a-886dbf9b0e72" />
<img width="1131" alt="Image" src="https://github.com/user-attachments/assets/5342315f-97f2-4372-af7d-6645b5ecc06e" />
<img width="1131" alt="Image" src="https://github.com/user-attachments/assets/a99ee5c0-4ace-4bc9-be04-22824eb396da" />
<img width="1131" alt="Image" src="https://github.com/user-attachments/assets/c3f57a9f-f35a-4bfb-99ca-636914381e11" />

**Comment by InduPriya1805** [comment]
Rev:Inse

**Comment by tj-devel709** [comment]
Note for self, it looks like the Label for the trash icon below disappears when the IconImageSource is set. With this removed, the Label is set even without the SemanticProperties.Description

```
    <ContentPage.ToolbarItems>
        <ToolbarItem
            Text="Delete"
            Command="{Binding DeleteCommand}"
            Order="Primary"
            Priority="0"
            IconImageSource="{StaticResource IconDelete}"
            SemanticProperties.Description="Delete" />
    </ContentPage.ToolbarItems>     
```

**Comment by AnKushSingh05** [comment]
Issue is fixed. 

<img width="1105" alt="Image" src="https://github.com/user-attachments/assets/ec91030d-a2a8-4531-b98f-d0a35e225e91" />
<img width="1105" alt="Image" src="https://github.com/user-attachments/assets/789bbe6a-ea3b-435d-9ed2-049793f477f7" />
